pulumi/grapl/__main__.py

```diff
@@ -159,6 +160,11 @@ def main() -> None:
             rust_log="DEBUG",
         )
 
+        ConsulIntention(
+            "grapl-core",
+            intention_directory=Path("../../nomad/intentions").resolve(),
+        )
```

Comment:
As I'm reading this code, the fact that the class is called `ConsulIntention`, but the intentions themselves are stored in a `nomad` directory makes me think that this is somehow a copy/paste error. If I were looking for where our Consul intentions are defined, I would instinctively look for a `consul` directory; it would take a while before I thought to look for them in a `nomad` directory, even though they are intentions that govern the behavior of our Nomad-managed services.

One solution would be to rename the directory `consul`. Another could be to rename `intentions` to `consul-intentions`.

On top of either of those, it may also be worth renaming `ConsulIntention` to something like `GraplServiceIntentions`.

Reply:
Added a comment regarding not needing an explicit provider. Basically its the same reason that the NomadJobs in local don't need one. Pulumi will use the consul:address if its set in the config or use the default localhost:8500 address. We need an explicit provider for the AWS section because we're setting the address from the StackReference in some cases.

Reply:
> As I'm reading this code, the fact that the class is called `ConsulIntention`, but the intentions themselves are stored in a `nomad` directory makes me think that this is somehow a copy/paste error. If I were looking for where our Consul intentions are defined, I would instinctively look for a `consul` directory; it would take a while before I thought to look for them in a `nomad` directory, even though they are intentions that govern the behavior of our Nomad-managed services.
> 
> One solution would be to rename the directory `consul`. Another could be to rename `intentions` to `consul-intentions`.
> 
> On top of either of those, it may also be worth renaming `ConsulIntention` to something like `GraplServiceIntentions`.

I've gone down the route of renaming the existing folder to consul-intentions. I believe that if we want people to remember to update the intention rules, it'll be easier if the config files are colocated. Medium-term I'd want to add enforcement and autogeneration to make this easier.

I also think that its worth keeping the explicit `ConsulIntentions` name so that new hires know that `intentions` are a consul construct and not a grapl-specific one.